Add GDScript warning pages (#10635)
godotengine/godot-docs: tutorials/scripting/gdscript/warnings/unsafe_call_argument.rst

UNSAFE_CALL_ARGUMENT
========================

The warning message is:

.. code-block:: none

    The argument 1 of the function "set_health()" requires the subtype "int" but the supertype "Variant" was provided.

The default warning level for this warning is **Ignore**.
To modify it, see :ref:`ProjectSettings.debug/gdscript/warnings/unsafe_call_argument<class_ProjectSettings_property_debug/gdscript/warnings/unsafe_call_argument>`.

When this warning occurs
------------------------

This warning may appear when passing a value of a certain type to a function that is expecting a subtype instead.

.. code-block::

    var original_health = 10

    func _ready():
        set_health(original_health)  # Will give warning UNSAFE_CALL_ARGUMENT.

    func set_health(new_health: int):
        current_health = new_health

> Because the type of ``original_health`` was not explicitly stated, it is inferred to be a ``Variant``. The ``Variant`` type is broader than the ``int`` type expected by the ``set_health()`` function, so it is possible that the variable ``original_health`` could hold a kind of value incompatible with it.

Reviewer: ```suggestion
Because the type of ``original_health`` was not explicitly stated, it is inferred to be a ``Variant``. The ``Variant`` type is broader than the ``int`` type expected by the ``set_health()`` function's parameter, so it is possible that the variable ``original_health`` could be an incompatible type.
```
or similar. I kind of understand why "kind of value" was said, so I am not sure if mine is the right, either.
